src/main/webapp/script.js

```diff
@@ -90,17 +94,39 @@ function drawRegionsMap(censusDataArray, title) {
 
 // createDataArray takes in the data array returned by the census API
 // and reformats it into a data table for the visualization API.
-function createDataArray(censusDataArray) {
+function createDataArray(censusDataArray, description) {
   const vizDataArray = [];
-  censusDataArray.forEach((state) => {
-    vizDataArray.push([state[0], state[1]/1000]);
-  });
-  // Changes the header of the vizDataArray to match the Visualization API
+  // If the length of the census data array is four, the census call has
```

Comment:
Is there a website link where you found this? Can you add that link here?

Reply:
Right now our website is programmed to take in states and counties. A county call has the header ["NAME","S0201_001E","state","county"] while a state call can either have the header ["NAME","S0201_154E","S0201_157E","state"] or ["NAME","S0201_154E","state"] depending on the type of call that is made. This isn't from a website it's just from my experience working with the limited queries thus far. It's also not a sure fire way to test for a percentage query, but it works with our limited set of queries.

Reply:
I am little confused. I thought you specify the variables you want in the request and the api returns appropriate values. Can you give me example queries where it returns 4 columns vs when it returns 3 columns.

Reply:
A query for "People of all ages who moved to each state" has the header ["NAME","S0201_119E","S0201_126E","state"]. We must do a calculation using the second and third columns of data to return the appropriate total. However, a query for "People of all ages who live in each state" has the header ["Name", "S0201_001E", "state"]. 

This is why I was testing the length, because the number of columns will indicate whether we have to do a percent to total calculation.

Reply:
ok it seems to be this should actually be happening in the backend which has all the information about the tables and fields etc. and the frontend should not have to do these kind manipulations. Let's talk about this in one of our meetings.

Reply:
Sounds good!